### DISCUSSION TITLE
Formattare le distanze chilometriche progressive

### DISCUSSION THREAD

**Discussion by fedekikko89** [opendatasicilia/HfcQGIS-md#79] [Idee]
Buongiorno,
desidero condividere con voi una formula analoga a quella contenuta nel vostro esempio svolto _"Formattare le distanze chilometriche progressive"_ al link https://hfcqgis.opendatasicilia.it/esempi/distanze_progressive_chilometriche/ , ma più sintetica e generale, nel caso in cui voleste integrarla nella vostra guida:

La formula è la seguente:

_floor("prog"/1000)||'+'||lpad("prog"-floor("prog"/1000)*1000,'3','0')_

Il risultato che si ottiene è lo stesso di quello dell'esercizio riportato al superiore link con la differenza che funziona (teoricamente) anche oltre i 1000 km, però non credo che (praticamente) esistano strade così lunghe... :-)

Un saluto

Federico

**Comment by pigreco** [comment]
Grazie mille @fedekikko89 
appena posso la integro nell'esempio svolto https://hfcqgis.opendatasicilia.it/esempi/distanze_progressive_chilometriche/

saluti
